src/librustc_middle/ty/util.rs

```diff
@@ -778,6 +778,58 @@ impl<'tcx> ty::TyS<'tcx> {
         }
     }
 
+    /// Returns `true` if equality for this type is both total and structural.
```

Comment:
The "total" comes from "totally ordered". Not sure if this terminology extends to equality.

Reply:
I don't know what universally standard terminology is here, but in our own docs for [`PartialEq`](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html), the adjective "full" is used to differentiate, as in "full equivalence relation" versus "partial equivalence relation."

Reply:
(Or you could say "if equality for this type is both reflexive and structural", since the only thing differentiating an equivalence relation from a partial equivalence relation (PER) is whether the relation is reflexive.